packages/flutter/lib/src/material/app.dart

```diff
@@ -35,62 +35,52 @@ const TextStyle _errorTextStyle = const TextStyle(
 ///  * [WidgetsApp]
 ///  * [Scaffold]
 ///  * [MaterialPageRoute]
-class MaterialApp extends WidgetsApp {
+class MaterialApp extends StatefulWidget {
+
   /// Creates a MaterialApp.
   ///
   /// At least one of [home], [routes], or [onGenerateRoute] must be
   /// given. If only [routes] is given, it must include an entry for
   /// the [Navigator.defaultRouteName] (`'/'`).
   ///
-  /// See also the [new WidgetsApp] constructor (which this extends).
+  /// This class creates an instance of [WidgetsApp].
   MaterialApp({
     Key key,
-    String title,
+    this.title,
     ThemeData theme,
-    Widget home,
-    Map<String, WidgetBuilder> routes: const <String, WidgetBuilder>{},
-    RouteFactory onGenerateRoute,
-    LocaleChangedCallback onLocaleChanged,
+    this.home,
+    this.routes: const <String, WidgetBuilder>{},
+    this.onGenerateRoute,
+    this.onLocaleChanged,
     this.debugShowMaterialGrid: false,
-    bool showPerformanceOverlay: false,
-    bool showSemanticsDebugger: false,
-    bool debugShowCheckedModeBanner: true
+    this.showPerformanceOverlay: false,
+    this.showSemanticsDebugger: false,
+    this.debugShowCheckedModeBanner: true
   }) : theme = theme,
-       home = home,
-       routes = routes,
-       super(
-    key: key,
-    title: title,
-    textStyle: _errorTextStyle,
-    color: theme?.primaryColor ?? Colors.blue[500], // blue[500] is the primary color of the default theme
-    onGenerateRoute: (RouteSettings settings) {
-      WidgetBuilder builder = routes[settings.name];
-      if (builder == null && home != null && settings.name == Navigator.defaultRouteName)
-        builder = (BuildContext context) => home;
-      if (builder != null) {
-        return new MaterialPageRoute<Null>(
-          builder: builder,
-          settings: settings
-        );
-      }
-      if (onGenerateRoute != null)
-        return onGenerateRoute(settings);
-      return null;
-    },
-    onLocaleChanged: onLocaleChanged,
-    showPerformanceOverlay: showPerformanceOverlay,
-    showSemanticsDebugger: showSemanticsDebugger,
-    debugShowCheckedModeBanner: debugShowCheckedModeBanner
-  ) {
+       textStyle = _errorTextStyle,
+       color = theme?.primaryColor ?? Colors.blue[500], // blue[500] is the primary color of the default theme
```

Comment:
color can literally just be read from the theme now that we can can hold it in a build function, rather than needing crazy antics in the constructor